### PR TITLE
Add signalr args: 'service_mode' and 'allowed_origins'

### LINKED ISSUE #4422: service-mode and allowed-origins arguments missing from azurerm_signalr_service
## Community Note ##
- Please vote on this issue by adding a 👍 reaction to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment
## Description ##
It would be really helpful if "allowed-origins" and "service-mode" are also included. Their absence introduces extra manual effort.

## References ##
https://docs.microsoft.com/en-us/cli/azure/signalr?view=azure-cli-latest

### PR DESCRIPTION
**Pull request by brunhil**
Fixes #4422 by adding service_mode and allowed_origins to the azurerm_signalr_service. 

Needed to update the signalr api version as the preview version did not have these fields present. Added acceptance tests verifying as well.

### CODE REVIEW

File: website/docs/r/signalr_service.html.markdown [hunk 1]
```diff
@@ -52,6 +65,20 @@ A `sku` block supports the following:
 
 * `capacity` - (Required) Specifies the number of units associated with this SignalR service. Valid values are `1`, `2`, `5`, `10`, `20`, `50` and `100`.
 
+---
+
+A `features` block supports the following:
+
+* `flag` - (Required) A kind of feature. e.g. ServiceMode.
```
Comment (tombuildsstuff) [inline website/docs/r/signalr_service.html.markdown]: to match the other resources could we make this:

```suggestion
* `flag` - (Required) A kind of feature. At this time the only supported value is `ServiceMode`.
```

File: website/docs/r/signalr_service.html.markdown [hunk 2]
```diff
@@ -52,6 +65,20 @@ A `sku` block supports the following:
...
+
+* `value` - (Required) A value of a feature flag. e.g. Default, Serverless, or Classic.
```
Comment (tombuildsstuff) [inline website/docs/r/signalr_service.html.markdown]: to match the other resources could we make this:

```suggestion
* `value` - (Required) A value of a feature flag. Possible values are `Classic`, `Default` and `Serverless`.
```

File: website/docs/r/signalr_service.html.markdown [hunk 3]
```diff
@@ -42,6 +51,10 @@ The following arguments are supported:
 
 * `sku` - A `sku` block as documented below.
 
+* `features` - (Optional) A `features` block as documented below.
+
+* `cors` - (Optional) A `cors` block as documented below.
```
Comment (tombuildsstuff) [inline website/docs/r/signalr_service.html.markdown]: _minor_ could we sort these alphabetically? e.g. `cors` -> `features` -> `sku`?

**Comment by brunhil** [comment]
@tombuildsstuff  - Addressed your comments, thanks for the feedback! I resolved your comments as well, let me know if this is okay to do.

File: azurerm/resource_arm_signalr_service.go [hunk 4]
```diff
@@ -72,6 +72,48 @@ func resourceArmSignalRService() *schema.Resource {
 				},
 			},
 
+			"features": {
+				Type:     schema.TypeList,
+				Optional: true,
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_signalr_service.go]: running the tests it appears this is also defaulted by Azure:

```
------- Stdout: -------
=== RUN   TestAccAzureRMSignalRService_capacityUpdate
=== PAUSE TestAccAzureRMSignalRService_capacityUpdate
=== CONT  TestAccAzureRMSignalRService_capacityUpdate
--- FAIL: TestAccAzureRMSignalRService_capacityUpdate (247.31s)
    testing.go:569: Step 0 error: After applying this step, the plan was not empty:
        
        DIFF:
        
        UPDATE: azurerm_signalr_service.test
          cors.#:                      "1" => "0"
          cors.0.allowed_origins.#:    "1" => ""
          cors.0.allowed_origins.0:    "*" => ""
          features.#:                  "1" => "0"
          features.0.flag:             "ServiceMode" => ""
          features.0.value:            "Default" => ""
          hostname:                    "acctestsignalr-191025150234218152.service.signalr.net" => "acctestsignalr-191025150234218152.service.signalr.net"
          id:                          "/subscriptions/1a6092a6-137e-4025-9a7c-ef77f76f2c02/resourceGroups/acctestRG-191025150234218152/providers/Microsoft.SignalRService/SignalR/acctestSignalR-191025150234218152" => "/subscriptions/1a6092a6-137e-4025-9a7c-ef77f76f2c02/resourceGroups/acctestRG-191025150234218152/providers/Microsoft.SignalRService/SignalR/acctestSignalR-191025150234218152"
          ip_address:                  "13.93.42.15" => "13.93.42.15"
          location:                    "westeurope" => "westeurope"
          name:                        "acctestSignalR-191025150234218152" => "acctestSignalR-191025150234218152"
          primary_access_key:          "oU+NldB5H13VX8+ZMsB2hPZskelTQYaAdRFGkikhI78=" => "oU+NldB5H13VX8+ZMsB2hPZskelTQYaAdRFGkikhI78="
          primary_connection_string:   "Endpoint=https://acctestsignalr-191025150234218152.service.signalr.net;AccessKey=oU+NldB5H13VX8+ZMsB2hPZskelTQYaAdRFGkikhI78=;Version=1.0;" => "Endpoint=https://acctestsignalr-191025150234218152.service.signalr.net;AccessKey=oU+NldB5H13VX8+ZMsB2hPZskelTQYaAdRFGkikhI78=;Version=1.0;"
          public_port:                 "443" => "443"
          resource_group_name:         "acctestRG-191025150234218152" => "acctestRG-191025150234218152"
          secondary_access_key:        "wVi4jHPQ7D8ana5Mh5Rd8Bp166/xsKYRHkgOvU1Zueo=" => "wVi4jHPQ7D8ana5Mh5Rd8Bp166/xsKYRHkgOvU1Zueo="
          secondary_connection_string: "Endpoint=https://acctestsignalr-191025150234218152.service.signalr.net;AccessKey=wVi4jHPQ7D8ana5Mh5Rd8Bp166/xsKYRHkgOvU1Zueo=;Version=1.0;" => "Endpoint=https://acctestsignalr-191025150234218152.service.signalr.net;AccessKey=wVi4jHPQ7D8ana5Mh5Rd8Bp166/xsKYRHkgOvU1Zueo=;Version=1.0;"
          server_port:                 "443" => "443"
          sku.#:                       "1" => "1"
          sku.0.capacity:              "1" => "1"
          sku.0.name:                  "Standard_S1" => "Standard_S1"
```

.. as such we'll need to make this:

```suggestion
				Optional: true,
				Computed: true,
```

File: azurerm/resource_arm_signalr_service.go [hunk 5]
```diff
@@ -72,6 +72,48 @@ func resourceArmSignalRService() *schema.Resource {
...
+								"Classic",
+							}, false),
+						},
+					},
+				},
+			},
+
+			"cors": {
+				Type:     schema.TypeList,
+				Optional: true,
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_signalr_service.go]: running the tests it appears this is also defaulted by Azure:

```
------- Stdout: -------
=== RUN   TestAccAzureRMSignalRService_capacityUpdate
=== PAUSE TestAccAzureRMSignalRService_capacityUpdate
=== CONT  TestAccAzureRMSignalRService_capacityUpdate
--- FAIL: TestAccAzureRMSignalRService_capacityUpdate (247.31s)
    testing.go:569: Step 0 error: After applying this step, the plan was not empty:
        
        DIFF:
        
        UPDATE: azurerm_signalr_service.test
          cors.#:                      "1" => "0"
          cors.0.allowed_origins.#:    "1" => ""
          cors.0.allowed_origins.0:    "*" => ""
          features.#:                  "1" => "0"
          features.0.flag:             "ServiceMode" => ""
          features.0.value:            "Default" => ""
          hostname:                    "acctestsignalr-191025150234218152.service.signalr.net" => "acctestsignalr-191025150234218152.service.signalr.net"
          id:                          "/subscriptions/1a6092a6-137e-4025-9a7c-ef77f76f2c02/resourceGroups/acctestRG-191025150234218152/providers/Microsoft.SignalRService/SignalR/acctestSignalR-191025150234218152" => "/subscriptions/1a6092a6-137e-4025-9a7c-ef77f76f2c02/resourceGroups/acctestRG-191025150234218152/providers/Microsoft.SignalRService/SignalR/acctestSignalR-191025150234218152"
          ip_address:                  "13.93.42.15" => "13.93.42.15"
          location:                    "westeurope" => "westeurope"
          name:                        "acctestSignalR-191025150234218152" => "acctestSignalR-191025150234218152"
          primary_access_key:          "oU+NldB5H13VX8+ZMsB2hPZskelTQYaAdRFGkikhI78=" => "oU+NldB5H13VX8+ZMsB2hPZskelTQYaAdRFGkikhI78="
          primary_connection_string:   "Endpoint=https://acctestsignalr-191025150234218152.service.signalr.net;AccessKey=oU+NldB5H13VX8+ZMsB2hPZskelTQYaAdRFGkikhI78=;Version=1.0;" => "Endpoint=https://acctestsignalr-191025150234218152.service.signalr.net;AccessKey=oU+NldB5H13VX8+ZMsB2hPZskelTQYaAdRFGkikhI78=;Version=1.0;"
          public_port:                 "443" => "443"
          resource_group_name:         "acctestRG-191025150234218152" => "acctestRG-191025150234218152"
          secondary_access_key:        "wVi4jHPQ7D8ana5Mh5Rd8Bp166/xsKYRHkgOvU1Zueo=" => "wVi4jHPQ7D8ana5Mh5Rd8Bp166/xsKYRHkgOvU1Zueo="
          secondary_connection_string: "Endpoint=https://acctestsignalr-191025150234218152.service.signalr.net;AccessKey=wVi4jHPQ7D8ana5Mh5Rd8Bp166/xsKYRHkgOvU1Zueo=;Version=1.0;" => "Endpoint=https://acctestsignalr-191025150234218152.service.signalr.net;AccessKey=wVi4jHPQ7D8ana5Mh5Rd8Bp166/xsKYRHkgOvU1Zueo=;Version=1.0;"
          server_port:                 "443" => "443"
          sku.#:                       "1" => "1"
          sku.0.capacity:              "1" => "1"
          sku.0.name:                  "Standard_S1" => "Standard_S1"
```

.. as such we'll need to make this:

```suggestion
				Optional: true,
				Computed: true,
```

File: azurerm/resource_arm_signalr_service_test.go [hunk 6]
```diff
@@ -398,6 +468,48 @@ resource "azurerm_signalr_service" "test" {
...
+  name                = "acctestSignalR-%d"
+  location            = "${azurerm_resource_group.test.location}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+
+  cors {
+	allowed_origins = [
+	  "http://www.contoso.com",
+	  "www.contoso.com",
+	  "contoso.com",
+	]
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_signalr_service_test.go]: this is missing a closing `}` here:

```suggestion
	]
  }
```

File: azurerm/resource_arm_signalr_service_test.go [hunk 7]
```diff
@@ -398,6 +468,48 @@ resource "azurerm_signalr_service" "test" {
...
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_signalr_service" "test" {
+  name                = "acctestSignalR-%d"
+  location            = "${azurerm_resource_group.test.location}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_signalr_service_test.go]: this is missing a `sku` block (which is required) could we add one here?

File: azurerm/resource_arm_signalr_service_test.go [hunk 8]
```diff
@@ -398,6 +468,48 @@ resource "azurerm_signalr_service" "test" {
...
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_signalr_service_test.go]: this is missing a `sku` block (which is required) could we add one here?

File: azurerm/resource_arm_signalr_service.go (hunk 5 above)
Comment (brunhil) [reply azurerm/resource_arm_signalr_service.go]: @tombuildsstuff - Will we also need to default the service mode setting as well? I don't have an intricate knowledge of how exactly _Computed_ setting works.

File: azurerm/resource_arm_signalr_service.go [hunk 9]
```diff
@@ -72,6 +72,50 @@ func resourceArmSignalRService() *schema.Resource {
 				},
 			},
 
+			"features": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_signalr_service.go]: since this is a Computed we can remove this since it's not needed:

```suggestion
```

**Comment by tombuildsstuff** [comment]
hey @brunhil 

Thanks for pushing those changes - I've spent a little while debugging this and it appears there's an error in the tests which needed to be fixed to get this to run - as such I hope you don't mind but so that we can get this merged I've pushed a commit which fixes this.

Thanks!

**Comment by tombuildsstuff** [comment]
Tests pass:

Screenshot 2019-10-29 at 12 30 44

**Pull request merged**